graphics/palette.h

```diff
@@ -111,9 +111,98 @@ class PaletteManager : Common::NonCopyable {
 
 namespace Graphics {
 
+struct Palette {
+	byte data[256 * 3];
+	uint size;
+
+	/**
+	 * @brief Construct a new Palette object
+	 *
+	 * @param num   the number of palette entries
+	 */
+	Palette(uint num = 0);
+
+	Palette(const Palette &p);
+
+	bool equals(const Palette &p) const;
```

Comment:
It might be nice to use operator overloads here.
```suggestion
	bool operator==(const Palette &p) const;
```

Reply:
Sure. I'm not normally a big fan of operator overloading, but this one makes enough sense.

Reply:
Equality operators added.